opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -252,6 +254,45 @@ namespace Opm
         }
     }
 
+    struct WriterCall
+    {
+        BlackoilOutputWriter& writer_;
+        std::unique_ptr< SimulatorTimerInterface > timer_;
+        const SimulatorState state_;
+        const WellState wellState_;
+        const bool substep_;
+
+        std::future< bool > asyncWait_;
```

Comment:
I did it this way because then the main thread does not have to wait on the output. It will simply spawn threads that wait if the previous thread was not finished. I have tried the other version too but it did not seem to give any performance improvement. I'll report back when I have some results.